scan/connectivity.go

```diff
@@ -50,10 +50,78 @@ func dnsLookupScan(host string) (grade Grade, output Output, err error) {
 	if len(addrs) == 0 {
 		err = errors.New("no addresses found for host")
 	}
+
 	grade, output = Good, addrs
 	return
 }
 
+var cloudflareNets []*net.IPNet
+
+func initOnCloudFlareScan() error {
+	// Don't re-download ranges if we already have them.
+	if len(cloudflareNets) > 0 {
+		return nil
+	}
+
+	// Download CloudFlare CIDR ranges and parse them.
+	v4resp, err := http.Get("https://www.cloudflare.com/ips-v4")
+	if err != nil {
+		return fmt.Errorf("Couldn't download CloudFlare IPs: %v", err)
+	}
+	defer v4resp.Body.Close()
+
+	v6resp, err := http.Get("https://www.cloudflare.com/ips-v6")
+	if err != nil {
+		return fmt.Errorf("Couldn't download CloudFlare IPs: %v", err)
+	}
+	defer v6resp.Body.Close()
+
+	scanner := bufio.NewScanner(io.MultiReader(v4resp.Body, strings.NewReader("\n"), v6resp.Body))
+	for scanner.Scan() {
+		_, ipnet, err := net.ParseCIDR(scanner.Text())
+		if err != nil {
+			return fmt.Errorf("Couldn't parse CIDR range: %v", err)
+		}
+		cloudflareNets = append(cloudflareNets, ipnet)
```

Comment:
We can also end up with a partial stored cloudflareNets if an error is encountered after only some of the ranges have been added.